torch/export/unflatten.py

```diff
@@ -1002,8 +1018,39 @@ def get_actual_output_node(output):
                 proxy_out.meta["val"] = orig_output.meta.get("val")
                 self.parent.node_map[orig_output] = proxy_out
 
-        if self.cached_graph_module is not None:
-            _verify_graph_equivalence(self.cached_graph_module, self.module)
+        num_calls: Dict[str, int] = {}
+        child_fqn = _call_name(self.fqn, self.num_calls(self.fqn))
+        accessor = _compute_accessor(self.parent.fqn, child_fqn)
+        deduplicated = False
+        # Iterate over all previously seen modules, and deduplicate if possible
+        for k, seen_module in self.seen_modules[self.module_id][:-1]:
+            num_calls[k] = num_calls.get(k, 0) + 1
+            seen_child_fqn = _call_name(k, num_calls[k])
+            if _check_graph_equivalence(seen_module, self.module):
```

Comment:
Does this mean that differently specialized graphs (e.g. `N(x, False), N(x, True)`) won't share state? As in if we do attribute swaps on `foo.bar`, it won't have the same change for `foo.bar@1` if the computation is different.

Reply:
I think they will share state because the same params / buffer objects have been assigned to all variants. See `assign_attr`.

Reply:
The intended effect is that these variants are like different methods on the same "moral" instance.

Reply:
Oh, I meant if someone were to modify params based on the original FQNs, like `foo.bar.attr = foo.bar.attr.bfloat16()`, `foo.bar@1.attr` won't see the same change? I've debugged some internal FSDP sharding pipelines that'll do this to modify parameter dtype at runtime. 

I think this falls under the same category of convenience APIs though, so no big deal

Reply:
Yeah I don't think it will see those changes unless they do it for all variants at the same time, so yeah, need that API. Any suggestions what that API should look like?

Reply:
For module swapping we can probably just patch the logic into Angela's `_swap_modules()` method in unflatten.py? For attributes probably similar: `def _swap_attributes(ep: ExportedProgram, attrs_to_swap: Dict[str, Union[Any, Callable[Any -> Any]]):`